numpy/core/tests/test_dtype.py

```diff
@@ -800,3 +802,16 @@ class PaddedStruct(ctypes.BigEndianStructure):
             ('b', '>H')
         ], align=True)
         self.check(PaddedStruct, expected)
+
+    all_types = set(''.join(typecodes.values()))
+    all_pairs = permutations(all_types, 2)
+
+    @pytest.mark.parametrize("pair", all_pairs)
+    def test_pairs(self, pair):
+        """
+        Check that np.dtype('x,y') matches [np.dtype('x'), np.dtype('y')]
+        Example: np.dtype('d,I') -> dtype([('f0', '<f8'), ('f1', '<u4')])
+        """
+        pair_type = [t[1] for t in np.dtype('{},{}'.format(*pair)).descr]
+        expected = list(map(lambda t: np.dtype(t).descr[0][1], pair))
```

Comment:
Weird to use `map` for one case and a list comprehension for the other.

Also, in my experience `.descr` is never what you want it to be

Reply:
Agree on `map` vs list comprehension, fixed.

Regarding `.descr` — it seems to be the only straightforward way of comparison between simple and complex `dtype`s. As far as I understood, `.descr` follows the [PEP3118](https://www.python.org/dev/peps/pep-3118/) requirements, and this is its only purpose. So it could be a reliable way of obtaining types.

Reply:
refactored the test